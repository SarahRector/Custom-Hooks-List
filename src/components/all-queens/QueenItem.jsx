import React from 'react';
import PropTypes from 'prop-types';

const QueenItem = ({ name, image_url }) => (
  <figure>
    <img src={image_url} alt={name} width="200"/>
    <figcaption>{name}</figcaption>
  </figure>
);

QueenItem.propTypes = {
  name: PropTypes.string.isRequired,
  image_url: PropTypes.string
};

export default QueenItem;
