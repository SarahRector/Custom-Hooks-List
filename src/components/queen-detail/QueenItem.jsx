import React from 'react';
import { useParams } from 'react-router-dom';
import { useQueenById } from '../../hooks/queens';

const QueenItem = () => {
  const { id } = useParams();
  const { loading, queen } = useQueenById(id);
  if(loading) return <h1>Loading...</h1>;
  console.log(queen);

  return (
    <>
      <h1>{queen.name}</h1>
      <img src={queen.image_url} alt={queen.name} />
      <p>{queen.quote}</p>
    </>
  );
};

export default QueenItem;
