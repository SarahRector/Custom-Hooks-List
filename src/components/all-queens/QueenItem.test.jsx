import React from 'react';
import { render, cleanup } from '@testing-library/react';
import QueenItem from './QueenItem';

describe('QueenItem component', () => {
  afterEach(() => cleanup());
  it('renders QueenItem', () => {
    const { asFragment } = render(<QueenItem
      name="Trixie"
      image_url="test.png" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
