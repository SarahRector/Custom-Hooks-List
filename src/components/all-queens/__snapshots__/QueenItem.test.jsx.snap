// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueenItem component renders QueenItem 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Trixie"
      src="test.png"
      width="200"
    />
    <figcaption>
      Trixie
    </figcaption>
  </figure>
</DocumentFragment>
`;
